ports/nrf/common-hal/bleio/Adapter.c

```diff
@@ -39,17 +39,34 @@
 #include "supervisor/usb.h"
 #include "shared-bindings/bleio/Adapter.h"
 #include "shared-bindings/bleio/Address.h"
+#include "mpconfigboard.h" // for BOARD_HAS_CRYSTAL
 
 STATIC void softdevice_assert_handler(uint32_t id, uint32_t pc, uint32_t info) {
     mp_raise_msg_varg(&mp_type_AssertionError,
                       translate("Soft device assert, id: 0x%08lX, pc: 0x%08lX"), id, pc);
 }
 
+static inline bool board_has_crystal(void) {
+#ifdef BOARD_HAS_CRYSTAL
+    return BOARD_HAS_CRYSTAL == 1;
+#else
+    return false;
+#endif
+}
+
 STATIC uint32_t ble_stack_enable(void) {
-    nrf_clock_lf_cfg_t clock_config = {
+    nrf_clock_lf_cfg_t clock_config;
+    if (board_has_crystal()) {
+        clock_config = (nrf_clock_lf_cfg_t){
         .source = NRF_CLOCK_LF_SRC_XTAL,
-        .accuracy = NRF_CLOCK_LF_ACCURACY_20_PPM
-    };
+        .accuracy = NRF_CLOCK_LF_ACCURACY_20_PPM};
+    } else {
+        clock_config = (nrf_clock_lf_cfg_t){
+        .source = NRF_CLOCK_LF_SRC_RC,
+        .rc_ctiv = 16,
+        .rc_temp_ctiv = 2,
+        .accuracy = NRF_CLOCK_LF_ACCURACY_250_PPM};
```

Comment:
hmmm, https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.s140.api.v6.1.1%2Fstructnrf__clock__lf__cfg__t.html states that accuracy "Must be 0 if source is not NRF_CLOCK_LF_SRC_RC."

Reply:
I don't see that accuracy needs to be 0 i only see that rc_ctiv and rc_temp_ctiv need to be 0.
Anyway the source in this case is NRF_CLOCK_LF_SRC_RC so this is legal. This specific config is also being used in the Adafruit nrf52 bootloader and according to the documentation these are the recommended values for this configuration.

Reply:
I read the infocenter information wrong, thanks for correcting me.